wfe/wfe.go

```diff
@@ -1586,14 +1575,6 @@ func (wfe *WebFrontEndImpl) Authz(
 	logEvent *requestEvent,
 	response http.ResponseWriter,
 	request *http.Request) {
-
-	// Only allow GET when not being strict
-	if request.Method == "GET" && wfe.strict {
-		response.Header().Set("Allow", "POST")
-		wfe.sendError(acme.MethodNotAllowed(), response)
-		return
-	}
-
 	authzID := strings.TrimPrefix(request.URL.Path, authzPath)
 	authz := wfe.db.GetAuthorizationByID(authzID)
```

Comment:
Maybe unrelated to this PR but it looks like we're fetching the authz (& returning a 404 if it isn't found) before validating the POST. I think it would be better to do the opposite.